src/board/v3/nap/nap_common.c

```diff
@@ -119,3 +124,64 @@ static void nap_exti_thread(void *arg)
   }
 }
 
+u32 nap_conf_rd_version(void)
+{
+  NAP->CONTROL = (NAP->CONTROL & ~((u32)NAP_CONTROL_VERSION_ADDR_Msk));
+  return NAP->VERSION;
+}
+
+u8 nap_conf_rd_version_string(char version_string[])
+{
+  u8 i = 0;
+  u32 reg = 0;
+  u32 ctrl = (NAP->CONTROL & ~((u32)NAP_CONTROL_VERSION_ADDR_Msk));
+
+  do {
+    NAP->CONTROL = ctrl | ((i + NAP_VERSION_STRING_OFFSET) <<
+        NAP_CONTROL_VERSION_ADDR_Pos);
+    reg = NAP->VERSION;
+    memcpy(version_string + i * sizeof(reg), &reg, sizeof(reg));
+    ++i;
+  } while (reg && i < NAP_VERSION_STRING_LENGTH);
+  version_string[i * sizeof(reg)] = 0;
+
+  return strlen(version_string);
+}
+
+void nap_rd_dna(u8 dna[])
+{
+  u32 reg = 0;
+  u32 ctrl = (NAP->CONTROL & ~((u32)NAP_CONTROL_VERSION_ADDR_Msk));
+
+  for (u8 i = 0; i < NAP_DNA_LENGTH; ++i) {
+    NAP->CONTROL = ctrl | ((i + NAP_DNA_OFFSET)
+        << NAP_CONTROL_VERSION_ADDR_Pos);
+    reg = NAP->VERSION;
+    memcpy(dna + i * sizeof(reg), &reg, sizeof(reg));
+  }
+}
+
+void nap_rd_dna_callback(u16 sender_id, u8 len, u8 msg[], void* context)
+{
+  (void)sender_id; (void)len; (void)msg; (void) context;
+  u8 dna[8];
+  nap_rd_dna(dna);
```

Comment:
This is not thread-safe due to the writes to `NAP->CONTROL`. Can we copy the DNA to a static variable once and use that instead?